### PR TITLE
Implemented React Router DOM for handing routing, added sidebar menu for navigation, and configured Ant Design dark theme globally

### PR DESCRIPTION
**Pull request by dipto-itransition**
### This PR introduces significant enhancements to our application, aimed at improving navigation, user interface consistency, and overall user experience. The key changes are as follows:

**Implemented React Router DOM:**
- Added React Router DOM to manage routing within the application.
- Set up routes for two pages: Home and Bookings.
- Ensured seamless navigation between these pages.

**Home Page & Bookings page**
-  Created Home.tsx under the pages directory & Moved existing content from App.tsx to a new Home page component & Added placeholder text ("Home")
-Created Bookings.tsx under the pages directory, Added placeholder text ("Bookings") as an initial implementation, setting up for future feature expansion.

**Static Side Menu:**
- Developed SilderMenu.tsx under the components/shared directory.
- Introduced a static left side menu for easy navigation across different pages.
- The menu includes links to the Home and Bookings pages & Included navigation links using Menu.Item components with HomeOutlined and BookOutlined icons.
- Utilized Ant Design's Menu and Sider components for a consistent look and feel.

**Ant Design Dark Theme:**
- Configured and applied Ant Design's dark theme globally.
- Ensured all components adhere to the dark theme for a cohesive and appealing user interface.
- Detailed Changes:

**Testing and Verification:**
- Added tests for Layout and Home components to ensure they render correctly.
- Verified navigation between Home and Bookings pages.
- Ensured the static side menu functions as expected.
- Confirmed the dark theme is applied consistently across all components.

Please review these changes and provide your feedback. Thank you for your attention and collaboration!

Best regards,
Dipto Dey

### CODE REVIEW

**Review by mainul-itransition — APPROVED** [review]
Maybe you missed some commented code in the app.text.tsx. Please remove those codes.

File: src/components/layout/layout.tsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+import { Layout as LayoutBase } from 'antd';
+import { LayoutProps } from '@/types/router';
+import SilderMenu from '../shared/SilderMenu';
```
Comment (alexeylos) [inline src/components/layout/layout.tsx]: I guess you meant `SiderMenu` here. Looks like just typo

File: src/components/shared/SilderMenu.tsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+const SilderMenu = () => {
+  const navigate = useNavigate();
+  const [currentLocation, setCurrentLocation] = useState(window.location.pathname);
+
+  const menuItems = [
+    { key: ROUTES.HOME, icon: <HomeOutlined />, label: 'Home' },
+    { key: ROUTES.BOOKINGS, icon: <BookOutlined />, label: 'Bookings' },
+  ];
+
+  const onClick: MenuProps['onClick'] = (e) => {
```
Comment (alexeylos) [inline src/components/shared/SilderMenu.tsx]: It's common approach to use `e` for events but let's avoid short aliases for function parameters. `event.key` for example is much more readable than `e.key`. So there's nothing wrong here but I suggest to use full names

File: src/components/shared/SilderMenu.tsx [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    <div className="custom-sider">
+      <Sider>
+        <div className="web-logo">
+          <img src="https://i.ibb.co/Xbn3Bdm/logo1.png" alt="website logo" />
+        </div>
+
+        <Menu
+          theme="dark"
+          onClick={onClick}
+          defaultOpenKeys={['sub1']}
```
Comment (alexeylos) [inline src/components/shared/SilderMenu.tsx]: What's `sub1` ?

**Review by alexeylos — APPROVED** [review]

**Pull request merged**